### PR TITLE
Implements pools

### PR DESCRIPTION
**Pull request by chipit24**
Related issue: #2. Note: This PR only implements pool functionality; lanes will come in another PR.

* If there are currently elements on the page but no pools, dragging a pool onto the page will move all elements into the new pool.
* Dragging elements around in the pool will expand the pool.
* If there are pools on the page, elements must be dragged onto the pool, otherwise it won't be added to the diagram.

What's currently not working:
* Parsing pools in XML ("Upload XML").
* Dragging elements between pools (currently just expands the pool).

### CODE REVIEW

File: src/components/Modeler.vue [hunk 1]
```diff
@@ -262,6 +317,41 @@ export default {
...
+      }
+
+      /* If there are any pools on the grid, only allow dragging components over pools */
+      if (!this.collaboration || this.collaboration.get('participants').length === 0) {
+        return;
+      }
+
+      const { x, y } = this.dragPoint;
+      if (clientX === x && clientY === y) {
+        /* We don't need to re-calcaulte values if mouse position hasn't changed  asdasdsd */
```
Comment (tuantruong625) [inline src/components/Modeler.vue]: Typo ?

**Comment by tdondich** [comment]
After deleting a pool, I can no longer drag items over the canvas.

Steps to reproduce:
- Create a pool
- Drag start event into pool
- Delete start event
- Delete pool
- Attempt to drag control into canvas

**Comment by velkymx** [comment]
I need to be able to resize the pool to add new elements.

**Comment by chipit24** [comment]
> I need to be able to resize the pool to add new elements.

@velkymx Currently, the pool expands when you drag an element near the edge, but this functionality will be modified (to allow dragging elements between lanes and pools). But I will make it so pools and lanes will still resize if you drag elements near the edge.

Resizing elements sounds like a task that should be handled in a different PR (and possibly different issue), to:
 * Cut down on merge conflicts.
 * Keep PRs smaller for quicker review and merge.
 * More easily track progress.

**Comment by tdondich** [comment]
@chipit24 appears to be an issue.   Reproduction steps:

- Drag a start event
- Drag a pool. Start event properly goes into initial lane of pool
- Create the lane crown element to create a new lane
- click on lane start event is in
- Notice start event is no longer visible

**Pull request merged**